exec/command.go

```diff
@@ -0,0 +1,186 @@
+// Package exec runs external commands
+package exec
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+	"syscall"
+)
+
+var (
+	// DefaultDebugfFn is the default debug print function.
+	DefaultDebugfFn = func(string, ...interface{}) { return }
+	// DefaultDebugPrefix is the default prefix that is prepended to messages passed to the debugf function.
+	DefaultDebugPrefix = "exec: "
+)
+
+// ExitCodeError is returned from Run() when a command exited with a code != 0.
+type ExitCodeError struct {
+	*Result
+}
+
+// Error returns the error description.
+func (e ExitCodeError) Error() string {
+	return fmt.Sprintf("exec: running '%s' in directory '%s' exited with code %d, expected 0, output: '%s'",
+		e.Command, e.Dir, e.ExitCode, e.Output)
+}
+
+// Cmd represents a command that can be run.
+type Cmd struct {
+	path string
+	args []string
+
+	dir           string
+	debugfFn      func(format string, v ...interface{})
+	debugfPrefix  string
+	expectSuccess bool
+}
+
+// Command returns a new Cmd struct.
+// If name contains no path separators, Command uses LookPath to
+// resolve name to a complete path if possible. Otherwise it uses name directly
+// as Path.
+// By default a command is run in the current working directory.
+func Command(name string, arg ...string) *Cmd {
+	return &Cmd{
+		path:         name,
+		args:         arg,
+		dir:          ".",
+		debugfFn:     DefaultDebugfFn,
+		debugfPrefix: DefaultDebugPrefix,
+	}
+}
+
+// ShellCommand executes a command in sh shell.
+func ShellCommand(cmd string) *Cmd {
+	return Command("sh", "-c", cmd)
```

Comment:
For baur's purpose I don't know a reason why somebody would want to use a different shell.

`ShellCommand` is only used for the build command to not having to either:
-  parse and convert the arguments in the build-command string to an args slice or 
- have the user to define them in an array in the config instead of one simple string.

Maybe there is an easy way to ensure correct argument handling and having not to run the command in a shell instead, that I'm not aware of? :-)

From the package's view and re-usability I agree that it would be nice to have it configurable.